app/controllers/webapp/Profile.scala

```diff
@@ -247,31 +247,108 @@ class Profile @Inject() (
+      //case Some(user) => Future.successful(WebAppResult.Ok(request, "profile.requestNVM.success", Nil, "Profile.requestNVM.success", Json.obj("status" -> "in progress")).getResult)
       case None => Future.successful(WebAppResult.Unauthorized(request, "error.noAuthenticatedUser", Nil, "AuthProvider.Identity.Unauthorized", Map[String, String]()).getResult)
     }
   }
-  
+
   def checkNVM = UserAwareAction.async { implicit request =>
     request.identity match {
       //dummy function. Need Validation from requestNVM for check if a user is NVM
-      case Some(user) => Future.successful(WebAppResult.Ok(request, "profile.checkNVM.success", Nil, "Profile.checkNVM.success", Json.obj("status" -> "denied", "conditions" -> Json.obj("hasAddress" -> false, "hasPrimaryCrew" -> false, "isActive" -> false))).getResult)
-      //case Some(user) => userService.checkNVM(user).map(_ match {
-	//case Some(nvmState) => WebAppResult.Ok(request, "profile.requestNVM.success", Nil, "Profile.requestNVM.success", Json.obj(nvmState)).getResult
-	//case None => WebAppResult.NotFound(request, "profile.requestNVM.notFound", Nil, "Profile.requestNVM.notFound", Map[String, String]()).getResult
-      //})
+      //case Some(user) => Future.successful(WebAppResult.Ok(request, "profile.checkNVM.success", Nil, "Profile.checkNVM.success", Json.obj("status" -> "denied", "conditions" -> Json.obj("hasAddress" -> false, "hasPrimaryCrew" -> false, "isActive" -> false))).getResult)
+      case Some(user) => {
+        user.profiles.headOption match {
+          case Some(profile) => userService.checkNVM(profile).map(nvmState =>
+            WebAppResult.Ok(request, "profile.checkNVM.success", Nil, "Profile.checkNVM.success", nvmState).getResult
+          )
+          case None => Future.successful(WebAppResult.NotFound(request, "profile.checkNVM.notFound", Nil, "Profile.checkNVM.notFound", Map[String, String]()).getResult)
+        }
+      }
+      case None => Future.successful(WebAppResult.Unauthorized(request, "error.noAuthenticatedUser", Nil, "AuthProvider.Identity.Unauthorized", Map[String, String]()).getResult)
+    }
+  }
+
+  /**
+    * Active flag
+    * The controller will check if a user is an active supporter of his main crew.
+    * Store the user in a list for Network-ASP to check if the request is valid
+    */
+  def checkActiveFlag = UserAwareAction.async { implicit request =>
+    request.identity match {
+      //dummy function. Need Validation from requestNVM for check if a user is NVM
+      //case Some(user) => userService.checkActiveFlag()
+      case Some(user) => {
+        user.profiles.headOption match {
+          case Some(profile) => userService.checkActiveFlag(profile).map(activeState =>
+            WebAppResult.Ok(request, "profile.checkActiveFlag.success", Nil, "Profile.checkActiveFlag.success", activeState).getResult
```

Comment:
Same here

Reply:
UserService returns a json. ATM I don't exaclty know how I could change this